inc/customizer/layouts.php

```diff
@@ -0,0 +1,498 @@
+	 */
+	public function load_post() {
+
+		if ( $this->meta_box ) {
+
+			add_action( 'add_meta_boxes', array( $this, 'add_meta_box' ), 10, 2 );
+
+		}
+
+	}
+
+	/**
+	 * Add post meta box for custom layouts.
+	 *
+	 * @see   $this->load_post()
+	 * @since 1.0.0
+	 * @uses  $this->render_meta_box()
+	 *
+	 * @param string  $post_type
+	 * @param WP_Post $post
+	 */
+	public function add_meta_box( $post_type, WP_Post $post ) {
+
+		if (
+			! post_type_supports( $post_type, 'primer-layouts' )
+			||
+			! current_user_can( 'edit_post_meta', $post->ID )
+			||
+			! current_user_can( 'add_post_meta', $post->ID )
+			||
+			! current_user_can( 'delete_post_meta', $post->ID )
+		) {
+
+			return;
+
+		}
+
+		add_meta_box(
+			'primer-layouts-meta-box',
+			esc_html__( 'Layout', 'primer' ),
+			array( $this, 'render_meta_box' ),
+			$post_type,
+			'side',
+			'default'
+		);
+
+	}
+
+	/**
+	 * Display the custom layouts post meta box.
+	 *
+	 * @see   $this->add_meta_box()
+	 * @since 1.0.0
+	 *
+	 * @param WP_Post $post
+	 * @param string  $box
+	 */
+	public function render_meta_box( WP_Post $post, $box ) {
+
+		$has_custom = ! empty( $this->get_post_layout( $post->ID ) );
+		$global     = $this->get_global_layout();
+
+		wp_nonce_field( basename( __FILE__ ), 'primer-layout-nonce' );
+
+		?>
+		<div class="primer-layout">
+
+			<p>
+			<?php
+
+			printf(
+				_x( 'The site-wide Default layout setting is located in the %s.', 'link to the Customizer', 'primer' ),
+				sprintf(
+					'<a href="%s">%s</a>',
+					admin_url( 'customize.php' ),
+					esc_html__( 'Customizer' )
+				)
+			);
+
+			?>
+			</p>
+
+			<p>
+				<input type="radio" name="primer-layout-override" id="primer-layout-use-default" value="0" <?php checked( ! $has_custom ) ?>><label for="primer-layout-use-default"><?php _e( 'Default', 'primer' ) ?></label>
+				&nbsp;&nbsp;
+				<input type="radio" name="primer-layout-override" id="primer-layout-use-custom" value="1" <?php checked( $has_custom ) ?>><label for="primer-layout-use-custom"><?php _e( 'Custom', 'primer' ) ?></label>
+			</p>
+
+			<div class="primer-layout-wrap">
+
+				<ul>
+
+					<?php foreach ( $this->layouts as $layout => $label ) :
+
+						$class = ( $layout === $global ) ? 'active global' : '';
+
+						if ( ! $class ) {
+
+							$class = ( $has_custom ) ? 'active' : 'disabled';
+
+						}
+
+						?>
+						<li class="<?php echo esc_attr( $class ) ?>">
+							<label for="primer-layout-<?php echo esc_attr( $layout ) ?>">
+								<input type="radio" name="primer-layout" id="primer-layout-<?php echo esc_attr( $layout ) ?>" value="<?php echo esc_attr( $layout ) ?>" <?php checked( $this->get_current_layout( $post->ID ), $layout ) ?> <?php disabled( 'disabled' === $class ) ?>>
+								<img src="<?php echo esc_url( sprintf( '%s/assets/layouts/%s%s.png', get_template_directory_uri(), $layout, is_rtl() ? '-rtl' : '' ) ) ?>" alt="<?php echo esc_attr( $label ) ?>" title="<?php echo esc_attr( $label ) ?>">
```

Comment:
@fjarrett I'm thinking svg might be better here since it's compatible almost everywhere. What do you think?